subprojects/core/src/main/java/org/gradle/groovy/scripts/internal/DefaultScriptCompilationHandler.java

```diff
@@ -281,24 +279,24 @@ public URL loadGroovySource(String filename) throws MalformedURLException {
     }
 
     private class CustomCompilationUnit extends CompilationUnit {
-        public CustomCompilationUnit(CompilerConfiguration compilerConfiguration, CodeSource codeSource, final Action<? super ClassNode> customVerifier, GroovyClassLoader groovyClassLoader) {
+        public CustomCompilationUnit(final ScriptSource script, CompilerConfiguration compilerConfiguration, CodeSource codeSource, final Action<? super ClassNode> customVerifier, GroovyClassLoader groovyClassLoader) {
             super(compilerConfiguration, codeSource, groovyClassLoader);
             this.verifier = new Verifier() {
                 public void visitClass(ClassNode node) {
                     customVerifier.execute(node);
                     super.visitClass(node);
                 }
             };
-            final GradleResolveVisitor resolveVisitor = new GradleResolveVisitor(this, defaultImportPackages, simpleNameToFQN);
+            final GradleResolveVisitor resolveVisitor = new GradleResolveVisitor(this, simpleNameToFQN);
             this.resolveVisitor = resolveVisitor;
             this.progressCallback = new ProgressCallback() {
                 @Override
                 public void call(ProcessingUnit context, int phase) throws CompilationFailedException {
                     if (phase == Phases.CANONICALIZATION) {
                         Set<String> deprecatedImports = resolveVisitor.getDeprecatedImports();
                         if (!deprecatedImports.isEmpty()) {
-                            DeprecationLogger.nagUserWith("Using " + Joiner.on(",").join(deprecatedImports)
-                                + " from the private org.gradle.util package without an explicit import is deprecated. Please either stop using these private classes (recommended) or import them explicitly at the top of your build file (not recommended). The implicit import will be removed in Gradle 5.0");
+                            DeprecationLogger.nagUserWith(StringUtils.capitalize(script.getDisplayName()) + " is using " + Joiner.on(" and ").join(deprecatedImports)
```

Comment:
Looks like we need one more test with other script name, i.e. `apply 'other.gradle'` and check default import in `other.gradle` .

Reply:
Done